core/meta/object-descriptor.js

```diff
@@ -405,34 +393,35 @@ var ObjectDescriptor = exports.ObjectDescriptor = Montage.specialize( /** @lends
         }
     },
 
-    _preparePropertyDescriptorsCache: {
-        value: function () {
-            var ownDescriptors = this._ownPropertyDescriptors,
-                isReady = true,
-                descriptor, i, n;
+_preparePropertyDescriptorsCache: {
+    value: function () {
+        var ownDescriptors = this._ownPropertyDescriptors,
+            isReady = true,
+            descriptor, i, n;
+        
+        if (!this._propertyDescriptorsAreCached)  {
             
-            if (!this._propertyDescriptors)  {
-                
+            for (i = 0, n = ownDescriptors.length; i < n && isReady; ++i) {
+                descriptor = ownDescriptors[i];
+                isReady = !!(descriptor && descriptor.name);
+            }
+
+            if (isReady) {
+                this._propertyDescriptorsAreCached = true;
+                this._propertyDescriptors = [];
+                this._propertyDescriptorsTable.clear();
                 for (i = 0, n = ownDescriptors.length; i < n && isReady; ++i) {
```

Comment:
iReady has to be true here, otherwise that code wouldn't be executed since there's a if(isReady) around, so probably can be removed?

Reply:
Yep. Good eye. That looks like a copy/paste error.